src/madness/world/worldrmi.cc

```diff
@@ -332,10 +335,19 @@ namespace madness {
     void RMI::RmiTask::huge_msg_handler(void *buf, size_t /*nbytein*/) {
         const size_t* info = (size_t *)(buf);
         int nword = HEADER_LEN/sizeof(size_t);
-        int src = info[nword];
-        size_t nbyte = info[nword+1];
-
-        RMI::task_ptr->hugeq.push_back(std::make_pair(src,nbyte));
+        const int src = info[nword];
+        const size_t nbyte = info[nword+1];
+        const int tag = info[nword+2];
+
+        // extra dose of paranoia: assert that we never process so many huge messages
+        // that the tag wraparound somewhere becomes possible ...
+        // the worst case is where only one node sends huge messages to every node in the communicator
+        // AND it has enough threads to use up all tags
+        // NB list::size() is O(1) in c++11, but O(N) in older libstdc++
+        MADNESS_ASSERT(ThreadPool::size() < RMI::RmiTask::unique_tag_period() ||
+                       RMI::task_ptr->hugeq.size() <
+                           RMI::RmiTask::unique_tag_period() / comm.Get_size());
```

Comment:
772684d43b73df3d9e832fc259f660edb4a2da06 introduced an RMI-specific communicator, and thus removes the tag collision issue (but I did introduce a separate range to keep thing kosher even without the change).

the check is correct: with nonpreemptive scheduler and nothing truly exotic happening on the programming side there is no way for 1 thread to work on 2 huge messages. So if # of threads < tag period the RMI tag wraparound cannot occur. If that is not the case we have to be very pessimistic (remember that sender has no way of checking how many tags on that node a live) ... think one node's threads firing roughly same number of huge messages to each other node

again: the test is correct in my opinion.